session/pingpong/consumer_totals_storage.go

```diff
@@ -29,20 +31,34 @@ const consumerTotalStorageBucketName = "consumer_promised_totals"
 type ConsumerTotalsStorage struct {
 	bolt persistentStorage
 	lock sync.Mutex
+	bus  eventbus.Publisher
 }
 
 // NewConsumerTotalsStorage creates a new instance of consumer totals storage.
-func NewConsumerTotalsStorage(bolt persistentStorage) *ConsumerTotalsStorage {
+func NewConsumerTotalsStorage(bolt persistentStorage, bus eventbus.Publisher) *ConsumerTotalsStorage {
 	return &ConsumerTotalsStorage{
 		bolt: bolt,
+		bus:  bus,
 	}
 }
 
 // Store stores the given amount as promised for the given channel.
 func (cts *ConsumerTotalsStorage) Store(consumerAddress, accountantAddress string, amount uint64) error {
```

Comment:
Probably can pass `consumerID` your validated value in upper layer, now it recalculated second time

Reply:
what?